featurebyte/service/feature_list.py

```diff
@@ -155,19 +168,22 @@ async def _extract_relationships_info(
         return relationships_info
 
     async def _update_features(
-        self, features: list[FeatureModel], feature_list_id: ObjectId
+        self,
+        features: list[FeatureModel],
+        inserted_feature_list_id: Optional[ObjectId] = None,
+        deleted_feature_list_id: Optional[ObjectId] = None,
     ) -> None:
-        feature_service = FeatureService(
-            user=self.user, persistent=self.persistent, catalog_id=self.catalog_id
-        )
         for feature in features:
-            await feature_service.update_document(
+            feature_list_ids = cast(List[ObjectId], feature.feature_list_ids)
+            if inserted_feature_list_id:
+                feature_list_ids = self.include_object_id(
```

Comment:
Does `include_object_id` remove duplicates?

Reply:
Good catch, seems like no. Will update this part.